pennylane_lightning_kokkos/src/simulator/StateVectorKokkos.hpp

```diff
@@ -756,6 +753,149 @@ template <class Precision> class StateVectorKokkos {
         }
     }
 
+    /**
+     * @brief Probabilities of each computational basis state.
+     *
+     * @return Floating point std::vector with probabilities
+     * in lexicographic order.
+     */
+    auto probs() -> std::vector<Precision> {
+        const size_t N = getLength();
+
+        Kokkos::View<Kokkos::complex<Precision> *> arr_data = getData();
+        Kokkos::View<Precision *> d_probability("d_probability", N);
+
+        // Compute probability distribution from StateVector using
+        // Kokkos::parallel_for
+        Kokkos::parallel_for(
+            Kokkos::RangePolicy<KokkosExecSpace>(0, N),
+            getProbFunctor<Precision>(arr_data, d_probability));
+
+        std::vector<Precision> probabilities(N, 0);
+
+        using UnmanagedPrecisionHostView =
+            Kokkos::View<Precision *, Kokkos::HostSpace,
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+        Kokkos::deep_copy(UnmanagedPrecisionHostView(probabilities.data(),
+                                                     probabilities.size()),
+                          d_probability);
+        return probabilities;
+    }
+
+    /**
+     * @brief Probabilities for a subset of the full system.
+     *
+     * @param wires Wires will restrict probabilities to a subset
+     * of the full system.
+     * @return Floating point std::vector with probabilities.
+     * The basis columns are rearranged according to wires.
+     */
+
+    auto probs(std::vector<size_t> wires) {
+        // Determining probabilities for the sorted wires.
+        const Kokkos::View<Kokkos::complex<Precision> *> arr_data = getData();
+        const size_t num_qubits = getNumQubits();
+
+        using UnmanagedSizeTHostView =
+            Kokkos::View<size_t *, Kokkos::HostSpace,
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+
+        using UnmanagedPrecisionHostView =
+            Kokkos::View<Precision *, Kokkos::HostSpace,
+                         Kokkos::MemoryTraits<Kokkos::Unmanaged>>;
+
+        using MDPolicyType_2D =
+            Kokkos::MDRangePolicy<Kokkos::Rank<2, Kokkos::Iterate::Left>>;
+
+        std::vector<size_t> sorted_ind_wires(std::move(wires));
+        const bool is_sorted_wires =
+            std::is_sorted(sorted_ind_wires.begin(), sorted_ind_wires.end());
+        std::vector<size_t> sorted_wires(sorted_ind_wires);
+
+        if (!is_sorted_wires) {
+            sorted_ind_wires = Util::sorting_indices(sorted_ind_wires);
+            for (size_t pos = 0; pos < wires.size(); pos++)
+                sorted_wires[pos] = wires[sorted_ind_wires[pos]];
+        }
```

Comment:
```suggestion
        const bool is_sorted_wires = std::is_sorted(wires.begin(), wires.end());
        std::vector<size_t> sorted_wires(wires);

        if (!is_sorted_wires) {
            auto sorted_ind_wires = Util::sorting_indices(wires);
            for (size_t pos = 0; pos < wires.size(); pos++)
                sorted_wires[pos] = wires[sorted_ind_wires[pos]];
        }
```
Hey @multiphaseCFD, I wonder if something like that makes sense.

Reply:
I didn't see @mlxd's comment.

Reply:
Thanks @AmintorDusko !